From 9eb0ba853a185a4063e411e5e8a1b566986ed244 Mon Sep 17 00:00:00 2001
From: Todd Poynor <toddpoynor@google.com>
Date: Tue, 12 Feb 2013 20:47:48 -0800
Subject: [PATCH] power: android-battery: push uevent whenever charge source
 changes

Ensure userspace reads an up-to-date value for charging status whenever the
charge source is updated.  Avoid races where stale charging state may be
reflected in userspace until the next battery state poll.

Change-Id: Ia744db48584e9e9abf442710c279af9e3a25c079
Signed-off-by: Todd Poynor <toddpoynor@google.com>
---
 drivers/power/android_battery.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/power/android_battery.c b/drivers/power/android_battery.c
index 6e8b405..8d45ff0 100644
--- a/drivers/power/android_battery.c
+++ b/drivers/power/android_battery.c
@@ -399,6 +399,7 @@ static void android_bat_charger_work(struct work_struct *work)
 
 	mutex_unlock(&android_bat_state_lock);
 	wake_lock_timeout(&battery->charger_wake_lock, HZ * 2);
+	power_supply_changed(&battery->psy_bat);
 }
 
 
-- 
1.8.0

